### PR TITLE
[enh] install-sd script for RPi

### PR DESCRIPTION
**Pull request by alexAubin**
This is a protoype of install-sd script for RPi, à la Internet Cube. Right now you can flash your SD card (with option `-f` given) and list RPi on local network

It's basically the same script as Internet Cube but with : 

- stuff related to Lime2, encryption and hypercube removed
- download RPi image instead of OLinuXino / Internet Cube
- `-l` option looks for RPi mac adresses instead of OLinuxino
- tweaked messages (s/internet cubes/raspberries)

Still needed : 

- change name of RPi image on build.yunohost.org
- should have an available GPG signature
- test all available options ?

Possible improvements : 

- have a mecanism similar to hypercube, such that you choose your post-installation arguments directly from this script, and post-installation is launched automatically at first boot.
- MacOS compatibility ?
- ???

### CODE REVIEW

File: scripts/install-sd-rpi.sh [hunk 1]
```diff
@@ -0,0 +1,615 @@
+#!/bin/bash
+
+# Raspberry Pi SD Card Installer
+# Inspired by :
```
Comment (alexAubin) [inline scripts/install-sd-rpi.sh:4]: Not sure what's the right stuff to put there

**Review by zamentur — APPROVED** [review]

**Review by zamentur — CHANGES_REQUESTED** [review]
GPG and renaming in yunohost-jessie-latest-sdraspi-stable.img.tar.xz needed

**Comment by zamentur** [comment]
change name of RPi image on build.yunohost.org -> Done

GPG check is missing

Result of my test:

> Continue? (yes/no) yes
> 
> [INFO] Option -s was not set, starting Assisted Block Device Detection
> 1. Please, remove the target SD card from your computer if present, then press Enter
> 2. Please, plug the target SD card into your computer (and don't touch to other devices), then press Enter
> 
> Block device /dev/mmcblk0 found. Use it as your SD card block device? (yes/no) yes
> [INFO] Option -s was set to /dev/mmcblk0
> [INFO] This script needs a sudo access
> [sudo] password for ljf: 
> [INFO] Downloading Debian/YunoHost image (HTTPS)
> [INFO] Image file: yunohost-jessie-latest-sdraspi-stable.img.tar.xz
> ######################################################################## 100,0%
> [INFO] Downloading GPG signature (HTTPS)
> ######################################################################## 100,0%
> [INFO] Checking GPG signature
> [ERR] GPG signature error

**Comment by alexAubin** [comment]
>change name of RPi image on build.yunohost.org -> Done

Thanks a lot !

I realize that the name for the image I put in the script ends up in .img.tar.xz , which is different from the .zip we have at the moment on https://build.yunohost.org .. Did you created a tar archive ? 

Also I was hoping that we could change the link to the RPi image on https://build.yunohost.org to something like `yunohost-jessie-YYYYMMDD-sdraspi-stable.img.tar.xz` with YYYYMMDD being the day the image was generated. And have `yunohost-jessie-latest-sdraspi-stable.img.tar.xz` pointing to the most recent image :stuck_out_tongue: .

**Comment by zamentur** [comment]
> Did you created a tar archive ?

Yes but I have not sign it

For the "latest" symlink, it's still possible

**Comment by alexAubin** [comment]
I pushed a refactoring of some parts of the script, intended to : 
- make a few things clearer (e.g. color codes)
- not systematically re-download the image (before, it was downloaded to a tmp directory, deleted after each execution ?)
- management of custom arguments (e.g. image path) was not easy to understand
- we don't have GPG signature for now, but we have a checksum for integrity, so I refactored all the part about validating the image file.

So far I have not tested all the possible behaviour when feeding a custom image name though.

**Review by zamentur — APPROVED** [review]
LGTM (read and tested)
I suggest to create an other pr to merge this script with the internetcube one

**Comment by zamentur** [comment]
Could be merged in one week (9 june) or the 5 june if there is an other approval

**Pull request merged**